de.metas.adempiere.adempiere/base/src/main/java/org/adempiere/bpartner/service/IBPartnerBL.java

```diff
@@ -150,4 +150,13 @@
 	 * @task https://github.com/metasfresh/metasfresh/issues/1090
 	 */
 	I_C_BPartner createFromTemplate(I_C_BPartner_QuickInput template);
+
+	/**
+	 * Retrieves the discount schema for the given BParnter. If the BPartner has none, it falls back to the partner's C_BP_Group.
+	 * If the partner has no group or that group hasn't a discount schema either, it returns <code>-1</code>.
+	 *
+	 * @param soTrx if <code>true</code>, the sales discount schema is returned, otherwise the purchase discount schema is returned.
+	 * @return partner's discount schema or -1
+	 */
+	int getDiscountSchemaId(I_C_BPartner bpartner, boolean soTrx);
```

Comment:
just loud thinking: in future, maybe we should use an enum instead of the boolean parameter. Could save if the javadoc line

Reply:
Yes we could an enum.

About saving the javadoc, not sure... depends on how you name it.
If it's a generic SOTrx enum like { SALES, PURCHASE } then it won't.

If it's a specific enum, tailored only for discount schema(s) and maybe only for this method then we can name it something like DiscountSchemaSOTrx (very bad name, but u got the point) and yes, in that case we can save the javadoc.